server/resp/src/main/java/org/infinispan/server/resp/commands/string/SETRANGE.java

```diff
@@ -0,0 +1,65 @@
+package org.infinispan.server.resp.commands.string;
+
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+
+import org.infinispan.server.resp.ByteBufferUtils;
+import org.infinispan.server.resp.Consumers;
+import org.infinispan.server.resp.Resp3Handler;
+import org.infinispan.server.resp.RespCommand;
+import org.infinispan.server.resp.RespRequestHandler;
+import org.infinispan.server.resp.commands.Resp3Command;
+
+import io.netty.buffer.Unpooled;
+import io.netty.channel.ChannelHandlerContext;
+
+/**
+ * SETRANGE Resp Command
+ *
+ * @link https://redis.io/commands/setrange/
+ * @since 15.0
+ */
+public class SETRANGE extends RespCommand implements Resp3Command {
+   public SETRANGE() {
+      super(4, 1, 1, 1);
+   }
+
+   @Override
+   public CompletionStage<RespRequestHandler> perform(Resp3Handler handler,
+         ChannelHandlerContext ctx,
+         List<byte[]> arguments) {
+      if (arguments.size() != this.getArity() - 1) {
+         ByteBufferUtils.stringToByteBuf("-ERR wrong number of arguments for 'getrange' command\r\n",
+               handler.allocator());
+         return handler.myStage();
+      }
+      byte[] keyBytes = arguments.get(0);
+      int offset = Integer.parseInt(new String(arguments.get(1), StandardCharsets.US_ASCII));
+      byte[] patch = arguments.get(2);
+      CompletionStage<Long> objectCompletableFuture = handler.cache().getAsync(keyBytes)
+            .thenCompose(value -> applyRange(handler, keyBytes, value, patch, offset));
+      return handler.stageToReturn(objectCompletableFuture, ctx, Consumers.LONG_BICONSUMER);
+   }
+
+   private CompletionStage<Long> applyRange(Resp3Handler handler, byte[] key, byte[] value, byte[] patch, int offset) {
+      byte[] newValue = setrange(value, patch, offset);
+      return handler.cache().putAsync(key, newValue).thenApply(v -> (long)newValue.length);
+   }
+   private byte[] setrange(byte[] value, byte[] patch, int offset) {
+      if (value==null) {
+         value = new byte[0];
+      }
+      if (patch.length+offset < value.length) {
+         var buf = Unpooled.wrappedBuffer(value);
+         buf.writerIndex(offset);
+         buf.writeBytes(patch, 0, patch.length);
+         return buf.array();
+      }
+      var buf = Unpooled.buffer(offset+patch.length);
```

Comment:
Maybe we should wrap this ByteBuf and release it in case of an exception is thrown.

Reply:
wrapped in try/catch